src/shared/utilities/timeoutUtils.ts

```diff
@@ -27,29 +41,31 @@ export class TimeoutError extends Error {
  * @param timeoutLength Length of timeout duration (in ms)
  */
 export class Timeout {
```

Comment:
This class is very close to `CancellationTokenSource` now but with a built-in timer. So I think it's in a decent position.